internal/app/registry_service/repository/equipment.go

```diff
@@ -17,97 +19,95 @@ func NewEquipment(db *sqlx.DB) Equipment {
 	return Equipment{db: db}
 }
 
-func (eqr *Equipment) List() ([]model.Equipment, error) {
-	var eql []model.Equipment
-	err := eqr.db.Select(&eql, `SELECT id, type, status, parameters, created_at, updated_at FROM equipment`)
+func (repo *Equipment) List() ([]*dtos.EquipmentGet, error) {
+	var eqs []model.Equipment
+	err := repo.db.Select(&eqs, `SELECT id, kind, status, parameters, created_at, updated_at FROM equipment`)
 	if err != nil {
 		return nil, err
 	}
-	return eql, nil
-}
-
-func (eqr *Equipment) Create(et model.EquipmentType, ep model.Params) (uuid.UUID, error) {
-	// Marshal the Parameters map to JSON
-	jep, err := json.Marshal(ep)
-	if err != nil {
-		return uuid.UUID{}, err
+	count := len(eqs)
+	eqgs := make([]*dtos.EquipmentGet, count, count)
+	for i, eq := range eqs {
+		eqgs[i] = dtos.EquipmentGetFromModel(eq)
 	}
+	return eqgs, nil
+}
 
-	var id uuid.UUID
-	for {
+func (repo *Equipment) Create(eqc *dtos.EquipmentCreate) (uuid.UUID, error) {
+	for jParameters, _ := json.Marshal(eqc.Parameters); ; {
 		// Generate a UUID version 6 (using a library):
-		id, err = uuid.NewV6()
-		if err != nil {
-			return uuid.UUID{}, err
+		id, err := uuid.NewV6()
+		if err == nil {
+			_, err = repo.db.NamedExec(
+				`INSERT INTO equipment (id, kind, status, parameters) VALUES (:id, :kind, :status, :parameters)`,
+				map[string]interface{}{
+					"id":         id,
+					"kind":       eqc.Kind,
+					"status":     model.Operational,
+					"parameters": jParameters,
+				},
+			)
+			if err == nil {
+				return id, nil
+			}
+			if false {// TODO: Check id duplicate error
```

Comment:
if isDuplicate(id)

Reply:
Meant "check if the insertion has failed because of uuid collision". Still don't know how to check that.

Fixed confusing comment.